File: backend/src/api/dto/albums.py
"""Album-search DTOs - reusable across features, not game-specific (see api.py's
/albums/search). Mirrors api/dto/persons.py exactly."""

from uuid import UUID

from pydantic import BaseModel

from domain.album import Album


class AlbumSearchResultOut(BaseModel):
    id: UUID
    name: str

    @classmethod
    def from_album(cls, album: Album) -> "AlbumSearchResultOut":
        return cls(id=album.id, name=album.name)


class AlbumSearchOut(BaseModel):
    results: list[AlbumSearchResultOut]

    @classmethod
    def from_albums(cls, albums: list[Album]) -> "AlbumSearchOut":
        return cls(results=[AlbumSearchResultOut.from_album(a) for a in albums])

File: frontend/src/api/types/albums.ts
// Reusable across features (not just Albumdle's guess input) - mirrors backend/src/api/dto/
// albums.py - see backend/src/api/api.py's /albums/search.

export interface AlbumSearchResultOut {
  id: string
  name: string
}

export interface AlbumSearchOut {
  results: AlbumSearchResultOut[]
}
